packages/main/src/Avatar.ts

```diff
@@ -414,6 +430,71 @@ class Avatar extends UI5Element implements ITabbable, IAvatarGroupItem {
 
 		return ariaHaspopup;
 	}
+
+	_observeImageSlot() {
+		const imageSlot = this.shadowRoot?.querySelector("slot:not([name])") as HTMLSlotElement;
+		if (!imageSlot) {
+			return;
+		}
+
+		imageSlot.addEventListener("slotchange", this._handleImagesBound); // Event for future slot changes
+		this._handleImages(); // Immediate setup for existing images
+	}
+
+	_disconnectImageSlotObserver() {
+		const imageSlot = this.shadowRoot?.querySelector("slot:not([name])") as HTMLSlotElement;
+		if (imageSlot) {
+			imageSlot.removeEventListener("slotchange", this._handleImagesBound);
+		}
+		this._detachImageErrorHandler();
+	}
+
+	_attachImageErrorHandler() {
```

Comment:
This function attaches a load event as well but it's called _attachImage**Error**Handler, consider renaming it to _attachImageEventHandlers or something similar

It also holds multiple responsibilities: 
1) attaches handlers
2) checks for already loaded avatars and updates the state

you can move the logic for the check and state update into a dedicated method and call it in handleImages.